gitlab: 12.7.6 -> 12.8.1

talyz opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Update GitLab to the latest version.

https://about.gitlab.com/releases/2020/02/22/gitlab-12-8-released/
https://about.gitlab.com/releases/2020/02/24/gitlab-12-8-1-released/

This is basically #81057 with fixes and minor additions. I can't add anything to that PR, so I opened this instead.

This still suffers from #79374.

###### Things done
- Tested a few things locally in a VM
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

talyz committed:
gitlab: 12.7.6 -> 12.8.1

https://about.gitlab.com/releases/2020/02/22/gitlab-12-8-released/
https://about.gitlab.com/releases/2020/02/24/gitlab-12-8-1-released/

talyz committed:
gitaly: 1.83.0 -> 12.8.1

In order to build gitaly, this locally overrides the version of
libgit2, since gitaly is not compatible with the latest version.

talyz committed:
gitlab-workhorse: 8.20.0 -> 8.21.0

talyz committed:
gitlab-shell: Change name from gitlab-shell-go to gitlab-shell

This is left over from when gitlab-shell had a ruby part and a go
part. The ruby part is now gone, so let's call the go part
gitlab-shell.

talyz committed:
gitaly: Add myself to maintainers

talyz commented:
@GrahamcOfBorg test gitlab

petabyteboy commented on pkgs/applications/version-management/gitlab/gitaly/deps.nix:
```diff
@@ -1319,8 +1319,8 @@
       url = "https://github.com/ugorji/go";
-      rev = "d75b2dcb6bc8";
-      sha256 = "0di1k35gpq9bp958ywranpbskx2vdwlb38s22vl9rybm3wa5g3ps";
+      rev = "v1.1.4";
```
I'd like to get a quick confirmation from @flokli that this undoing this change from https://github.com/NixOS/nixpkgs/pull/80062 is fine.

talyz replied:
Hm, yeah, it seems a little strange..

petabyteboy replied:
Seems like it was not done on purpose so it should be fine

flokli replied:
As written there, I don't know why `vgo2nix` it moved to that rev (maybe another package wasn't compatible with it?!), but that rev seems to be older than 1.1.4.

If it now resolves to 1.1.4 again, works for me :-)

petabyteboy commented:
Interestingly this fixes #81636, so it would make #81506 unnecessary.

petabyteboy commented on pkgs/applications/version-management/gitlab/gitaly/default.nix:
```diff
@@ -16,15 +17,24 @@ let
         };
       };
   };
+  libgit2_0_27 = libgit2.overrideAttrs (oldAttrs: rec {
```
Can you add a note about this fix in the commit description?

talyz replied:
Sure!

talyz replied:
Should be fixed in my latest push.

talyz committed:
bundlerEnv: Add option to copy gem files instead of symlinking

The way ruby loads gems and keeps track of their paths seems to not
always work very well when the gems are accessed through
symlinks. Ruby will then complain that the same files are loaded
multiple times; it relies on the file's full path to determine whether
the file is loaded or not.

This adds an option to simply copy all gem files into the environment
instead, which gets rid of this issue, but may instead result in major
file duplication.

talyz committed:
gitaly: Copy gem files into bundler env instead of symlinking

This fixes issue #79374, where gitaly prints warning messages on the
client side when running push or fetch.

talyz commented:
I just added a fix for #79374. It touches the `bundlerEnv` code, but shouldn't affect any other packages than `gitaly` atm.

@GrahamcOfBorg test gitlab

petabyteboy commented:
I deployed the new version and can confirm it fixes https://github.com/NixOS/nixpkgs/issues/79374 too. Thanks!

petabyteboy commented:
I also ran nix-review to make sure the changes do bundlerEnv do not create problems for other ruby applications.
